### PR TITLE
gnu-style flags, environment variables, dockerfile

### PR DESCRIPTION
**Pull request by ryscheng**
- Adds a base Dockerfile for use and one for the Talek replica that is GPU compatible.
- Replaces use of 'flag' with 'pflag', which uses GNU-style flags
- Allow flags to be set via environment variables or command-line (e.g. TALEK_CONFIG=talek.conf go run main.go --socket=pirsocket)

### CODE REVIEW

File: cli/talekreplica/main.go [hunk 1]
```diff
@@ -49,6 +63,11 @@ func main() {
 		return
 	}
 
+	log.Printf("Using the following configuration:")
+	log.Printf("serverConfig=%#+v\n", serverConfig)
+	log.Printf("serverConfig.Config=%#+v\n", serverConfig.Config)
+	log.Printf("serverConfig.TrustDomain=%#+v\n", serverConfig.TrustDomain)
```
Comment (willscott) [inline cli/talekreplica/main.go]: shouldn't print out keymaterial

File: cli/talekutil/main.go [hunk 2]
```diff
@@ -3,33 +3,41 @@ package main
...
-var outputClient = flag.Bool("client", false, "Create configuration for a talek client.")
-var outputReplica = flag.Bool("replica", false, "Create configuration for a talek server.")
-var outputTD = flag.Bool("trustdomain", false, "Create raw trustdomain configuration.")
-var name = flag.String("name", "talek", "Server Name.")
-var address = flag.String("address", "localhost:9000", "Server Address.")
-var infile = flag.String("infile", "", "Begin with configuration from file.")
-var outfile = flag.String("outfile", "talek.json", "Save configuration to file.")
-var private = flag.Bool("private", false, "Include private key configuration.")
-var trustdomains = flag.String("trustdomains", "talek.json", "Comma separated list of trust domains.")
+const envPrefix = "TALEK"
```
Comment (willscott) [inline cli/talekutil/main.go]: this is duplicated everywhere. can it just be exported from common?

File: cli/talekreplica/main.go (hunk 1 above)
Comment (ryscheng) [inline cli/talekreplica/main.go]: Done

File: cli/talekutil/main.go (hunk 2 above)
Comment (ryscheng) [inline cli/talekutil/main.go]: Done

**Comment by ryscheng** [comment]
Updated to include your comments. I've been treating your `serverrefactor` branch kind of like master, since I didn't want to conflict too much. What else do we need to do to merge it all into master?

**Comment by willscott** [comment]
i need to look over your reviews - i think mostly around atomic safety things

**Pull request merged**